### DISCUSSION TITLE
UGS is confusing

### DISCUSSION THREAD

**Discussion by fygh1** [winder/Universal-G-Code-Sender#2403] [Help]
Hello,
I just intalled UGS but have hard time with it. 
1) Please, where is a new gcode tamplate file location so I can change it for my cnc station? So when I choose new gcode file, it will be prefilled as I want.
2) Editor is not allowing some GRBL command but I know they are valid GRBL 1.1 and my cnc station accepts it. But editor is broken and I do not know how to tell him to ignore all errors and just send all as plain text into cnc.
3) Where is setting for "before" and "after" macros? So that I do not have to copy the same code at the beginning and end of gcode file.

Thank you.

**Comment by breiler** [comment]
1. There is no such feature for setting the default template yet.
2. What GRBL command is that? The  error message is just a hint and you will be able to send the gcode anyway.
3. There is no such feature for predefined snippets yet.
